quickwit/quickwit-cli/src/main.rs

```diff
@@ -159,6 +168,87 @@ async fn main() -> anyhow::Result<()> {
     std::process::exit(return_code)
 }
 
+mod busy_detector {
+    use std::sync::atomic::{AtomicU64, Ordering};
+    use std::time::Instant;
+
+    use once_cell::sync::Lazy;
+    use tracing::warn;
+
+    // we need that time reference to use an atomic and not a mutex for LAST_UNPARK
+    static TIME_REF: Lazy<Instant> = Lazy::new(Instant::now);
+
+    const ALLOWED_DELAY_MICROS: u64 = 5000;
+    const WARN_SUPPRESSION_DURATION: u64 = 30_000_000;
+
+    thread_local!(static MY_ID: AtomicU64 = AtomicU64::new(0));
```

Comment:
If I understand correctly, MY_ID/NEXT_ID are here to assign a thread id to our thread. Why don't we just use the thread id?

Reply:
Getting the inner id is unstable, and `ThreadId` Debug representation is `ThreadId(1)` which isn't very pretty